ahrefs/monorobot
# handle status notification

## Description of the task

### Status Event
- Accept status events and check of the state of the status payload (success, failure, and error will be accepted, and pending will be ignored).
- Query the url that points to a commit in the status payload and parse the json.
- Filter status events based the prefix rules of the commit path and send notifications based on routing.
- The status notification includes include information about the state of a status, status description, commit, and author. 

### Other Minor changes
- Removed links to author/sender Git profile in all events (push, pull requests, issues, comments, and status).

## How to test
In mock_payloads, one can find newly added .json files called ```status.SOMETHING.json```, where SOMETHING contains the state and other information about a status event.

To test that notabot is now able to accept status events, run the following commands:
```
./notabot check --json path\to\some\file
```

Currently, tests are expected to fail as the querying a url may require a token, but we have temporarily included a dummy token in the code (https://github.com/ahrefs/notabot/blob/status/lib/github.ml#L69).

## References
- existing issue: fix #15
